### PR TITLE
8260941: Remove the conc_scan parameter for CardTable

### PR DESCRIPTION
**Pull request by tschatzl**
Hi,

  can I have reviews for this removal of the last(?) CMS-specific code in CardTable, namely some provision to indicate that cards are being scanned concurrently in Serial/Parallel GC barrier code?

The change simply follows the predicate into Serial/Parallel GC code which always returns false for them and removes that code.

In the review for JDK-8234534 I mentioned that I split this out due to unexplainable errors; testing tier1-5 three times showed none of that any more (after updating to latest code).

This change has only been built on Oracle-platforms and linux-x86 via github actions (https://github.com/tschatzl/jdk/actions/runs/539993964), so I would like to kindly ask maintainers of the others to compile and report issues (32 bit ARM, PPC etc).

Testing: tier1-5

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260941](https://bugs.openjdk.java.net/browse/JDK-8260941): Remove the conc_scan parameter for CardTable

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - Author) ⚠️ Review applies to 5db8cbe673b6e99e59c14e3dec397b972bfa308b
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to 5db8cbe673b6e99e59c14e3dec397b972bfa308b

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2425/head:pull/2425`
`$ git checkout pull/2425`

### CODE REVIEW

**Review by albertnetymk — APPROVED** [review]
A side note: it seems that `G1BarrierSet` is the only subclass of `CardTableBarrierSet`. Maybe it makes sense to merge them into one.

File: src/hotspot/share/gc/g1/g1BarrierSet.hpp [hunk 1]
```diff
@@ -53,6 +53,10 @@ class G1BarrierSet: public CardTableBarrierSet {
   G1BarrierSet(G1CardTable* table);
   ~G1BarrierSet() { }
 
+  bool card_mark_must_follow_store() const {
```
Comment (kimbarrett) [inline src/hotspot/share/gc/g1/g1BarrierSet.hpp:56]: Should be "virtual".  (Probably should be "override", but nobody's gotten around to vetting and proposing that change to the style guide.)

**Review by kimbarrett — APPROVED** [review]
Looks good.  Just the one minor nit about "virtual".

**Comment by tschatzl** [comment]
Thanks for your revies @albertnetymk @kimbarrett 

/integrate

**Pull request closed without merging**